src/Core/Auth/Identity/DuoUniversalPromptService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Bit.Core.Auth.Models;
+using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Settings;
+using Duo = DuoUniversal;
+
+namespace Bit.Core.Auth.Identity;
+
+/* 
+    PM-5156 addresses tech debt
+    Interface to allow for DI, will end up being removed as part of the removal of the old Duo SDK v2 flows.
+    This service is to support SDK v4 flows for Duo. At some time in the future we will need
+    to combine this service with the DuoWebTokenProvider and OrganizationDuoWebTokenProvider to support SDK v4.
+*/
+public interface IDuoUniversalPromptService
+{
+    Task<string> GenerateAsync(TwoFactorProvider provider, User user);
+    Task<bool> ValidateAsync(string token, TwoFactorProvider provider, User user);
+}
+
+public class DuoUniversalPromptService : IDuoUniversalPromptService
+{
+    private readonly ICurrentContext _currentContext;
+    private readonly GlobalSettings _globalSettings;
+
+    /// <summary>
+    /// Constructor for the DuoUniversalPromptService. Used to supplement v2 implementation of Duo with v4 SDK
+    /// </summary>
+    /// <param name="currentContext">used to fetch initiating Client</param>
+    /// <param name="globalSettings">used to fetch vault URL for Redirect URL</param>
+    public DuoUniversalPromptService(
+        ICurrentContext currentContext,
+        GlobalSettings globalSettings)
+    {
+        _currentContext = currentContext;
+        _globalSettings = globalSettings;
+    }
+
+    /// <summary>
+    /// Provider agnostic (either Duo or OrganizationDuo) method to generate a Duo Auth URL
+    /// </summary>
+    /// <param name="provider">Either Duo or OrganizationDuo</param>
+    /// <param name="user">self</param>
+    /// <returns>AuthUrl for DUO SDK v4</returns>
+    public async Task<string> GenerateAsync(TwoFactorProvider provider, User user)
+    {
+        if (!HasProperMetaData(provider)) return null;
```

Comment:
Our code style [requires braces for conditionals](https://contributing.bitwarden.com/contributing/code-style/csharp#control-blocks), there's a couple instances of this throughout the PR.